src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/Processors/LedgerTransactionMarkers/OriginLedgerTransactionMarkerProcessor.cs

```diff
@@ -91,16 +96,22 @@ public void VisitTransaction(CoreApiSdk.Model.CommittedTransaction committedTran
                 });
         }
 
-        if (committedTransaction.LedgerTransaction is CoreApiSdk.Model.UserLedgerTransaction)
+        var origin = committedTransaction.LedgerTransaction switch
         {
-            _ledgerTransactionMarkersToAdd.Add(
-                new OriginLedgerTransactionMarker
-                {
-                    Id = _context.Sequences.LedgerTransactionMarkerSequence++,
-                    StateVersion = stateVersion,
-                    OriginType = LedgerTransactionMarkerOriginType.User,
-                });
-        }
+            FlashLedgerTransaction => LedgerTransactionMarkerOriginType.Flash,
+            GenesisLedgerTransaction => LedgerTransactionMarkerOriginType.Genesis,
+            RoundUpdateLedgerTransaction => LedgerTransactionMarkerOriginType.RoundUpdate,
+            UserLedgerTransaction => LedgerTransactionMarkerOriginType.User,
+            _ => throw new ArgumentOutOfRangeException($"Unexpected ledger transaction type: {committedTransaction.LedgerTransaction.GetType()}"),
+        };
+
+        _ledgerTransactionMarkersToAdd.Add(
+            new OriginLedgerTransactionMarker
```

Comment:
I take it these markers aren't actually exposed? I'm slightly surprised we don't need to update the Gateway API? Or does it just match on `LedgerTransactionMarkerOriginType.User`?

Out of interest, how does it distinguish epoch changes from round updates?